util/src/main/java/tc/oc/pgm/util/material/Materials.java

```diff
@@ -1,19 +1,121 @@
 package tc.oc.pgm.util.material;
 
+import static org.bukkit.Material.*;
+import static tc.oc.pgm.util.bukkit.BukkitUtils.parse;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import java.util.Map;
+import java.util.Set;
 import org.bukkit.Bukkit;
-import org.bukkit.Effect;
 import org.bukkit.Location;
 import org.bukkit.Material;
-import org.bukkit.block.Banner;
-import org.bukkit.block.Block;
 import org.bukkit.block.BlockState;
 import org.bukkit.inventory.ItemStack;
-import org.bukkit.inventory.meta.BannerMeta;
 import org.bukkit.inventory.meta.ItemMeta;
-import org.bukkit.material.MaterialData;
-import tc.oc.pgm.util.block.BlockFaces;
+import tc.oc.pgm.util.bukkit.BukkitUtils;
 
 public interface Materials {
+  Map<Integer, Material> BY_ID =
+      new ImmutableMap.Builder<Integer, Material>() {
+        {
+          for (Material value : Material.values()) {
+            put(value.getId(), value);
+          }
+        }
+      }.build();
+
+  // Weapons
+  Material WOODEN_SWORD = parse("WOOD_SWORD", "WOODEN_SWORD");
+  Material GOLDEN_SWORD = parse("GOLD_SWORD", "GOLDEN_SWORD");
+  Material WOOD_AXE = parse("WOOD_AXE", "WOODEN_AXE");
+  Material GOLD_AXE = parse("GOLD_AXE", "GOLDEN_AXE");
+  Material WOOD_PICKAXE = parse("WOOD_PICKAXE", "WOODEN_PICKAXE");
+  Material GOLD_PICKAXE = parse("GOLD_PICKAXE", "GOLDEN_PICKAXE");
+  Material WOOD_SHOVEL = parse("WOOD_SPADE", "WOODEN_SHOVEL");
+  Material STONE_SHOVEL = parse("STONE_SPADE", "STONE_SHOVEL");
+  Material GOLD_SHOVEL = parse("GOLD_SPADE", "GOLDEN_SHOVEL");
+  Material IRON_SHOVEL = parse("IRON_SPADE", "IRON_SHOVEL");
+  Material DIAMOND_SHOVEL = parse("DIAMOND_SPADE", "DIAMOND_SHOVEL");
+  Material WOOD_HOE = parse("WOOD_HOE", "WOODEN_HOE");
+  Material GOLD_HOE = parse("GOLD_HOE", "GOLDEN_HOE");
+
+  Material SIGN_POST = parse("SIGN_POST", "LEGACY_SIGN_POST");
```

Comment:
This, or the logic surrounding this, will need modification when we go for version support. 
Example: `ACACIA_SIGN != LEGACY_SIGN_POST`

This is however fine for now.